cumulus/zombienet/zombienet-sdk-helpers/src/lib.rs

```diff
@@ -295,15 +295,26 @@ fn extract_relay_parent_storage_root(
 	}
 }
 
-pub async fn submit_extrinsic_and_wait_for_finalization_success<C: Config, S: Signer<C>>(
-	client: &OnlineClient<C>,
+/// Submits the given `call` as transaction and waits for it successful finalization.
+///
+/// The transaction is send as immortal transaction.
+pub async fn submit_extrinsic_and_wait_for_finalization_success<S: Signer<PolkadotConfig>>(
+	client: &OnlineClient<PolkadotConfig>,
 	call: &DynamicPayload,
 	signer: &S,
-) -> Result<(), anyhow::Error>
-where
-	<C::ExtrinsicParams as ExtrinsicParams<C>>::Params: Default,
-{
-	let mut tx = client.tx().sign_and_submit_then_watch_default(call, signer).await?;
+) -> Result<(), anyhow::Error> {
+	let mut extensions: <<PolkadotConfig as Config>::ExtrinsicParams as ExtrinsicParams<
+		PolkadotConfig,
+	>>::Params = Default::default();
+
+	extensions.4 = CheckMortalityParams::<PolkadotConfig>::immortal();
```

Comment:
I think this could also just be:

```rust
use subxt::config::polkadot::PolkadotExtrinsicParamsBuilder as Params;

let extensions = Params::new().immortal().build()
```

(see https://github.com/paritytech/subxt/blob/master/subxt/examples/tx_with_params.rs)

Reply:
Not sure, but I think that since [`subxt-core v0.41.0`](https://docs.rs/crate/subxt-core/0.41.0/source/src/config/default_extrinsic_params.rs#118) it should be `extensions.5`.
That's why approach by @jsdw seems more resilient.

Reply:
Ok, I think I got it. `subxt` is not taken from current workspace but reimported from zombienet-sdk ([line 14](https://github.com/paritytech/polkadot-sdk/pull/9362/files/d34efa1ed6850a70f95f75273e38e54a8ce0b93e#diff-6a768d445c8e9b3815ec09ee4f8ded9e549e2bd719a075dac45e6e1d179f1bffR14)), which is version `0.3.8` which uses [`subxt-0.38`](https://github.com/paritytech/zombienet-sdk/blob/v0.3.8/Cargo.toml#L49).